### ISSUE TITLE
Enable error bands for histograms

### ISSUE THREAD

**Issue by andrzejnovak** [scikit-hep/mplhep#26]

**Comment by andrzejnovak** [comment]
Hi @hamzahanif2210, this is not currently implemented. If you'd like to work on the feature, I'd ask you to give a quick sketch here on what API and look you'd imagine for it.

**Comment by danielhundhausen** [comment]
Hi @andrzejnovak , I'd be happy to work on this. As a standalone function something like this would work:

```
def errorband(
    H_band,  # value array if error band is symmetric / tuple of upper and lower value arrays
    bins,  # corresponding to bin edges of the hist values supplied in H_band
    label=None,
    edges=True,
    ax=None,
    **kwargs,
):
    # ax check
    if ax is None:
        ax = plt.gca()
    else:
        if not isinstance(ax, plt.Axes):
            raise ValueError("ax must be a matplotlib Axes object")

    # Construction of upper and lower boundries
    if isinstance(H_band, tuple):
        y_upper = H_band[0]
        y_lower = H_band[1]
    if isinstance(H_band, np.ndarray):
        y_upper = H_band
        y_lower = -y_upper

    error_band_args = { 
        "edges": bins, "facecolor": "none", "linewidth": 0,
        "alpha": .9, "color": "black", "hatch": "///"
    }   
    ax.stairs(y_upper, baseline=y_lower, label=label, **{**error_band_args, **kwargs})
```

Please let me know your thoughts.

**Comment by andrzejnovak** [comment]
Welcome, @danielhundhausen! I think I was originally thinking that this could be an option in `histplot` along the line of `histplot(histtype='band')`, where the band could be determined either by setting `yerr` or automatically from the existing weights methods. Is there a reason why you'd say it would be better as a separate function?

**Comment by danielhundhausen** [comment]
Thank you for your comments @andrzejnovak ! I made a PR #414 that includes the integration of errorbands in `histplot` when plotting with `histtype="fill"`, which is the standard use case, I think.
Regarding your question about making a separate function for the plotting - I think it reduces complexity of the code to split things into separate functions whenever possible. For the integration in `histplot` I have tried to follow the pattern of `errorbar`.
